client/app/configuration/deployment-maps/DeploymentMapTargetController.js

```diff
@@ -267,20 +247,39 @@ angular.module('EnvironmentManager.configuration').controller('DeploymentMapTarg
       return _.map(deploymentMap.Value.DeploymentTarget, 'ServerRoleName');
     }
 
-    function validateDataForSave() {
-      var valid = { result: true };
+    function validateDataForPuppetRole() {
+      if (vm.target.ASG.LaunchConfig.AMIPlatform == 'Linux' && typeof vm.target.ASG.LaunchConfig.PuppetRole == 'undefined') {
+        return modal.confirmation({
+          title: 'Linux AMI/Puppet Role Warning',
+          message: 'It looks like you have chosen a linux AMI and not specified a puppet role. This can lead to problems later on when you deploy to an environment. Please ask on the #linux-infra channel in slack if you need help defining one. Do you want to continue?'
+        });
```

Comment:
It would be nice to have this link directly to the Slack channel (like in `DeploymentDetailsModalController.js` vm.deploymentSupportSlack = "slack://channel?id=C1HCFJU67&team=T060F53FW"; )